### ISSUE TITLE
Different results using skorch and my training function

### ISSUE THREAD

**Issue by sliwy** [braindecode/braindecode#60]
I was trying to translate my codes to use `braindecode`. Previously I had a function `train` that took pytorch neural net and trained model. Now I wanted to replace this function with skorch classifier. I created an example that should use the same network, the same parameters and the same dataset and it gives different results (with my `train` function it gives 1.0 train accuracy after around 10-15 epoch instead of not even similar results after 50 epochs). On my dataset it does not work at all, so that is why I got interested in the skorch performance.
I'm stuck and I don't have any idea what causes this behavior. If anyone has some time and would like to help me here is a link to the python file with the example, maybe fresh look can help https://github.com/sliwy/braindecode/blob/strange_example/examples/skorch_slow_learning.py

![image](https://user-images.githubusercontent.com/11686357/73221418-13521100-4161-11ea-8ad0-3d0f649da8d5.png)
![image](https://user-images.githubusercontent.com/11686357/73222175-e56dcc00-4162-11ea-9b3a-8374284957bf.png)

**Comment by robintibor** [comment]
Great that you put up that example. This is exactly what we need at the moment to verify real use cases. A quick scan through the code I didn't notice any reason unfortunately (you took care of shuffling batch size etc.).
The accuracy_score at the very end of the script what does it output?

**Comment by sliwy** [comment]
One thing I didn't fix is `random_seed` (and I think we should freeze it) but for each run you can draw similar conclusions. I've run it again and the accuracy_score is 0.32 (validation: 0.29 from 1st epoch) in comparison to 0.45 from my function (validation 0.58 for epoch 21).

**Comment by sliwy** [comment]
Today I will submit an issue to skorch, to ask what is the difference between skorch training loop and the on defined by me. I saved numpy arrays to remove not needed data processing and include everything except model defininton in one file https://github.com/sliwy/braindecode/blob/strange_example/examples/skorch_ext_example.py
Let you know if I know anything new.

**Comment by sliwy** [comment]
I'm closing it as I've found the reason, in my `train` function there was a typo and I was setting model to train mode only at the beginning of training. After first validation step the model was in `eval` mode till the end of training.

**Comment by robintibor** [comment]
I see, great catch. Then it means that in your case it is actually better to train in eval mode?
